src/ophys_etl/modules/segmentation/modules/calculate_edges.py

```diff
@@ -0,0 +1,125 @@
+import argschema
+import time
+import networkx as nx
+import multiprocessing
+import tempfile
+import h5py
+from typing import Optional
+from pathlib import Path
+from functools import partial
+import matplotlib
+
+from ophys_etl.modules.segmentation.modules.schemas import \
+    CalculateEdgesInputSchema
+from ophys_etl.modules.segmentation.graph_utils import (
+        partition, edge_attributes, creation)
+from ophys_etl.modules.segmentation.qc import graph_plotting
+
+
+def pearson_edge_job(graph_path: Path, video_path: Path) -> nx.Graph:
+    graph = edge_attributes.add_pearson_edge_attributes(
+            nx.read_gpickle(graph_path),
+            video_path)
+    nx.write_gpickle(graph, graph_path)
+    return graph_path
+
+
+def filtered_pearson_edge_job(filter_fraction: float,
+                              graph_path: Path,
+                              video_path: Path) -> nx.Graph:
+    graph = edge_attributes.add_filtered_pearson_edge_attributes(
+            filter_fraction,
+            nx.read_gpickle(graph_path),
+            video_path)
+    nx.write_gpickle(graph, graph_path)
+    return graph_path
+
+
+def hnc_gaussian_edge_job(
+        neighborhood_radius: int,
+        graph_path: Path,
+        video_path: Path,
+        attribute_name: str = 'hnc_Gaussian',
+        filter_fraction: Optional[float] = None,
+        full_neighborhood: bool = False) -> nx.Graph:
+    graph = edge_attributes.add_hnc_gaussian_metric(
+            graph=nx.read_gpickle(graph_path),
+            video_path=video_path,
+            neighborhood_radius=neighborhood_radius,
+            filter_fraction=filter_fraction,
+            attribute_name=attribute_name,
+            full_neighborhood=full_neighborhood)
+    nx.write_gpickle(graph, graph_path)
+    return graph_path
+
+
+class CalculateEdges(argschema.ArgSchemaParser):
+    default_schema = CalculateEdgesInputSchema
+
+    def run(self):
+        self.logger.name = type(self).__name__
+        t0 = time.time()
+
+        # select which calculation to perform
+        if self.args['attribute_name'] == 'Pearson':
```

Comment:
I wonder how many more possible calculations we might end up adding? If we plan to add many more, it might be worthwhile to switch to a dictionary based function lookup instead of a growing list of `elif` clauses?

e.g.
```
job_lookup_table = dict()
job_lookup_table["Pearson"] = pearson_edge_job
...
edge_job = job_lookup_table[self.args['attribute_name']]
```

Reply:
I agree, this will get cumbersome if it grows. I don't anticipate more, at the moment.